_posts/2021-11-19-connecting-java-high-level-rest-client-with-opensearch-over-https.markdown

```diff
@@ -0,0 +1,152 @@
+---
+layout: post
+title:  "Connecting Java High-level REST client with OpenSearch over HTTPS"
+authors: 
+  - setiah
+date: 2021-11-19 01:01:01 -0700
+categories: 
+  - technical-post
+twittercard:
+  description: "This post walks through the steps to configure Java High-level REST client to connect with OpenSearch over HTTPS"
+---
+
+If you have ever used OpenSearch with a java application, you might have come across OpenSearch Java high-level REST client. Java high-level REST client provides OpenSearch APIs as methods, and makes it easier for a java application to interact with OpenSearch using request/response objects.
+
+In this blog, we will see how to configure the Java high-level REST client (a.k.a JHLRC) to connect with an OpenSearch cluster securely. We will set up a local OpenSearch server using the default distribution which comes with security plugin pre-installed, and use JHLRC client in a sample Java application to interact with the server over HTTPS. I have used an Ubuntu 18.04 machine, but it should work for other linux distributions as well.
+
+**Setup OpenSearch Server**
+
+Lets start with downloading the [latest OpenSearch linux distribution](https://opensearch.org/downloads.html) from [opensearch.org](http://opensearch.org/). I have used 1.1.0 version which is the latest available version at the time of this writing. Unzip the downloaded tar.gz file.
+
+```
+// Download the latest available version from opensearch.org. 
+`> wget https://artifacts.opensearch.org/releases/bundle/opensearch/1.1.0/opensearch-1.1.0-linux-x64.tar.gz
+`
+// Unzip it
```

Comment:
This really untars it :) I think those comments aren't neessary, I would just leave the 2 lines